sopel/db.py

```diff
@@ -160,18 +274,40 @@ def unalias_nick(self, alias):
         """
         alias = Identifier(alias)
         nick_id = self.get_nick_id(alias, False)
-        count = self.execute('SELECT COUNT(*) FROM nicknames WHERE nick_id = ?',
-                             [nick_id]).fetchone()[0]
-        if count <= 1:
-            raise ValueError('Given alias is the only entry in its group.')
-        self.execute('DELETE FROM nicknames WHERE slug = ?', [alias.lower()])
+        session = self.ssession()
+        try:
+            count = session.query(Nicknames) \
+                .filter(Nicknames.nick_id == nick_id) \
+                .count()
+            if count <= 1:
+                raise ValueError('Given alias is the only entry in its group.')
+            session.query(Nicknames).filter(Nicknames.slug == alias.lower()).delete()
+            session.commit()
+        except SQLAlchemyError:
+            session.rollback()
+            raise
+        finally:
+            session.close()
 
     def delete_nick_group(self, nick):
         """Removes a nickname, and all associated aliases and settings."""
         nick = Identifier(nick)
         nick_id = self.get_nick_id(nick, False)
-        self.execute('DELETE FROM nicknames WHERE nick_id = ?', [nick_id])
-        self.execute('DELETE FROM nick_values WHERE nick_id = ?', [nick_id])
+        session = self.ssession()
+        try:
+            count = session.query(Nicknames) \
+                .filter(Nicknames.nick_id == nick_id) \
+                .count()
+            if count <= 1:
+                raise ValueError('Given alias is the only entry in its group.')
```

Comment:
A nick group is allowed to have only one member/alias. This would prevent deleting ungrouped nicks.

Reply:
Are you sure? If you look at test_db.py#178, there's multiple associated with a group.

Reply:
I might be misunderstanding what you're getting at, I tried to reimplement the exact logic as before

Reply:
What was here before included no such check for the group size. It looks like you copied a chunk of `unalias_nick()`, which _does_ care about whether the group will be empty, into here.